### ISSUE TITLE
[Due for payment 2025-04-02] [$500] Chat - Mentions are not shown in the issue Expensify Card thread page header

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#54630]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.79-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team
**Component:** Chat Report View

## Action Performed:
1. Go to https://staging.new.expensify.com/home
2. Go to workspace settings - enable expensify card
3. Issue a card
4. Open workspace chat
5. Open the reply in thread for the issue Expensify Card thread message

## Expected Result:
Mentions must be shown in the header of the "issue Expensify Card" thread page and a double space must not be shown. 
The LHN and Header should always show the mention in the same format as the original message.

## Actual Result:
Mentions are not shown in the "issue Expensify Card" thread page header - instead a double space is shown.

Also, for other messages with mentions, we currently replace the mention with the display first name.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/c8ac5072-0f09-4848-8fdf-dbdbb6b01f65

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021873775509911824270</li>
        <li>Upwork Job ID: 1873775509911824270</li>
        <li>Last Price Increase: 2025-01-20</li>
<li>Automatic offers: </li>
<ul>
<li>daledah | Contributor | 105805358</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @ahmedGaber93</details>

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Chat - Mentions are not shown in expensify issue card thread page instead a double space shown

### What is the root cause of that problem?
For the thread header case when the user hasn't set firstName it will be empty string and we only fallback to the login if it doesn't exist (`??`) so the assignee part will be empty 
https://github.com/Expensify/App/blob/12e0941f859c61385baa4b05e7513b0bd04b1d1f/src/libs/ReportActionsUtils.ts#L1809
### What changes do you think we should make in order to solve the problem?

We need to use `||` instead to deal with empty string cases
```
    const assignee = shouldRenderHTML ? `<mention-user accountID="${assigneeAccountID}"/>` : assigneeDetails?.firstName || (assigneeDetails?.login ?? '');

```
### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

**Comment by ahmedGaber93** [comment]
Sorry for the delay, reviewing today.

**Comment by FitseTLT** [comment]
> Sorry for the delay, reviewing today.

Thx I have added [some](https://github.com/Expensify/App/issues/54630#issuecomment-2564034709) detail in the RCA to make it clear 👍

**Comment by ahmedGaber93** [comment]
@daledah you used a different case (invited user mention) in your proposal instead of this issue case, I think they not the same implementation. "invited user" display the `reportAction` message from BE, but card issue build the message in FE side by [getCardIssuedMessage](https://github.com/Expensify/App/blob/754817de78a65879a01e13da3caf2d90033dfc47/src/libs/ReportActionsUtils.ts#L1794) 

https://github.com/Expensify/App/blob/754817de78a65879a01e13da3caf2d90033dfc47/src/libs/ReportUtils.ts#L4086-L4088

**Comment by ahmedGaber93** [comment]
>  ![Screenshot 2025-01-03 at 7 51 51 AM](https://github.com/user-attachments/assets/461f17a0-a2d4-4705-a83c-43b041343659)


Based on the expected behavior in this [comment](https://github.com/Expensify/App/pull/49177#discussion_r1759671784), I see we should check for `firstName` first in the thread title and LHN.

I think @FitseTLT's solution can fix that `assigneeDetails?.firstName || (assigneeDetails?.login ?? '')`

**Comment by ahmedGaber93** [comment]
> I just wonder if we should fix the bug (invite/remove room member) in this issue or not. If yes we can raise the bounty, or we can handle it in the separate issue. What do you think?

I think this completely different root cause, and should fix separator by fix `expensify-common` then bump it in `E/App`.

CC @sakluger WDYT?

**Comment by sakluger** [comment]
I'd prefer to solve both problems in this issue if possible.

@ahmedGaber93 thanks for posting the expected behavior from the [other issue](https://github.com/Expensify/App/pull/49177#discussion_r1759671784). So in the case where display name exists, we should show First Name in the header and LHN instead of showing the mention. @mountiny can you confirm the expected behavior here in the case where Display Name doesn't exist? Should we show a mention in the header & LHN, or the user's email address, or something else?

**Comment by sakluger** [comment]
If they are totally separate problems with different fixes, then sure, we can solve them in separate issues.

I still want to make sure we fully understand the expected behavior. Specifically, when no display name exists, should the header have a clickable mention, or an unclickable mention, or something else?

**Comment by FitseTLT** [comment]
> I still want to make sure we fully understand the expected behavior. Specifically, when no display name exists, should the header have a clickable mention, or an unclickable mention, or something else?

@sakluger there is no other instance where we set mentions to be clickable in report header title I don't think we want to make an exception here.

**Comment by ahmedGaber93** [comment]
Waiting @mountiny confirmation https://github.com/Expensify/App/issues/54630#issuecomment-2569631556

**Comment by sakluger** [comment]
I chatted with @mountiny offline and we both agree that mentions should be unclickable in the header. But just to be safe, I'm checking with the wider team on expected behavior for when display name exists and doesn't exist: https://expensify.slack.com/archives/C03U7DCU4/p1736358665731809

**Comment by ahmedGaber93** [comment]
Waiting on https://github.com/Expensify/App/issues/54630#issuecomment-2578284944

**Comment by sakluger** [comment]
Sorry, I think we're close, but we are still hashing a few things out. Hopefully we'll have this resolved and ready to get to work on Monday.

**Comment by ahmedGaber93** [comment]
Not overdue, Waiting on #54630 (comment)

**Comment by FitseTLT** [comment]
[Updated](https://github.com/Expensify/App/issues/54630#issuecomment-2564034709)

**Comment by ahmedGaber93** [comment]
> we always want the LHN and Header to always show the mention in the same format as the original message (but not clickable).


@sakluger Just for confirmation, will the bounty $500 cover this case https://github.com/Expensify/App/issues/54630#issuecomment-2568748987 or will be fixed separately?

**Comment by sakluger** [comment]
> Just for confirmation, will the bounty $500 cover this case https://github.com/Expensify/App/issues/54630#issuecomment-2568748987 or will be fixed separately?

@ahmedGaber93 I was thinking it would cover both, and that we could use this issue to make sure that the anytime a mention is used in the header or LHN, it would show the actual mention string vs a first name. Do you think that's okay?

**Comment by ahmedGaber93** [comment]
@sakluger Yeah, it's okay. Thanks for the confirmation.

**Comment by ahmedGaber93** [comment]
I will complete the review today.

**Comment by ahmedGaber93** [comment]
@daledah your fix to the second issue looks well, but for the first issue, your fix is different from the logic is used on `Parser` [here](https://github.com/Expensify/App/blob/018df994db38cd5e3e3a04a8f71b58a33c049bbb/src/libs/Parser.ts#L35)


``` 
const assignee = shouldRenderHTML ? `<mention-user accountID="${assigneeAccountID}"/>` : `@${assigneeDetails?.login ?? ''}
````
For that, I think letting `Parser` to handle this [case](https://github.com/Expensify/App/blob/754817de78a65879a01e13da3caf2d90033dfc47/src/libs/ReportActionsUtils.ts#L1809) looks better.



``` 
const assignee = shouldRenderHTML ? `<mention-user accountID="${assigneeAccountID}"/>` : Parser.htmlToText(`<mention-user accountID="${assigneeAccountID}"/>`)}
````

but we still have a mismatch when using short mentions, in message we used it [here](https://github.com/Expensify/App/blob/3c588ad96da3e12dd5fe3340d1cee73c22396c39/src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.tsx#L44), but not in header and LHN. Could you please investigate it?

**Comment by FitseTLT** [comment]
@ahmedGaber93 I think combining the two unrelated problems is not correct. There was a clear problem initially in the OP and I have proposed the solution to it and @daledah clearly copied my solution recently and treating the two unrelated problems is not fair. I think we should create an other issue for the second problem with its relevant OP and testing steps to make things clear and proceed with it.

And regarding your comment above what is the advantage of using Parser while we could simply use Personal Details List as we were using previously?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by ahmedGaber93** [comment]
I am still able to reproduce on Staging, plus we increase the scope

![Image](https://github.com/user-attachments/assets/b03ca4c1-9143-4de8-84cb-d653d24faeeb)

**Comment by ahmedGaber93** [comment]
> I think combining the two unrelated problems is not correct. There was a clear problem initially in the OP and I have proposed the solution to it

@FitseTLT I know the scope is changed, and I will take care about that when selecting the proposal/proposals. We still discuss the potential fixes, and you can update your proposal if you want

**Comment by ahmedGaber93** [comment]
> And regarding your comment above what is the advantage of using `Parser` while we could simply use Personal Details List as we were using previously?

@FitseTLT I think using `personalDetails` directly will not cover cases like removing `@SMSDomain` for phone number and fallback to `@hidden`. Plus, using the same function to get the mention text looks easy to maintenance and confirm the results is exactly the same.

**Comment by ahmedGaber93** [comment]
> but we still have a mismatch when using short mentions, in message we used it [here](https://github.com/Expensify/App/blob/3c588ad96da3e12dd5fe3340d1cee73c22396c39/src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.tsx#L44), but not in header and LHN. Could you please investigate it?

@daledah the LHN case is a BE issue, you need to investigate the `Header` case. When user click replay to thread on message with a short mention, the header title should be also with a short mention.

**Comment by ahmedGaber93** [comment]
> I updated https://github.com/Expensify/App/issues/54630#issuecomment-2564181112 to cover this case.

@daledah thanks for the updates, I think we can move forward now.

**Comment by ahmedGaber93** [comment]
Based on https://github.com/Expensify/App/issues/54630#issuecomment-2587965444, we decided that we always want the LHN and Header to always show the mention in the same format as the original message.

For the LHN issue https://github.com/Expensify/App/issues/54630#issuecomment-2593458355, It displays the `lastMessageText` that come from backend, so it needs to fix in BE side (we have a closed issue for it here https://github.com/Expensify/App/issues/44735)

For the other issues, we can move forward @daledah's [proposal](https://github.com/Expensify/App/issues/54630#issuecomment-2564181112).

🎀 👀 🎀 C+ reviewed

**Comment by nkuoch** [comment]
I will fix the BE side in https://github.com/Expensify/Auth/pull/13912

**Comment by ahmedGaber93** [comment]
@daledah, please let me know when PR is ready for review.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by ahmedGaber93** [comment]
I am still able to reproduce on Staging, plus we increase the scope

![Image](https://github.com/user-attachments/assets/0ea94de4-68fb-47e4-9116-0e03cdffbc44)

**Comment by ahmedGaber93** [comment]
PRSs on progress. @daledah please let me know when the draft [PR](https://github.com/Expensify/expensify-common/pull/831) is being ready for review.

**Comment by sakluger** [comment]
@daledah thanks for the update. In the future, if there will be a delay for holiday or other time off, please leave a quick comment if possible just so we are aware. Thanks!

**Comment by ahmedGaber93** [comment]
We are waiting expensify-common version 2.0.119 to be published on [npm](https://www.npmjs.com/package/expensify-common?activeTab=versions) so we can bump it in E/App.

**Comment by ahmedGaber93** [comment]
waiting  on https://github.com/Expensify/App/pull/54037 that will bump expensify-common to be merged, then will complete my PR review.

**Comment by sakluger** [comment]
Summarizing payment on this issue:

Contributor: @daledah $250, paid via Upwork
Contributor+: @ahmedGaber93 $250, please request on Newdot after completing BZ checklist

**Comment by sakluger** [comment]
@daledah thank you for the correction. The Upwork contract was correct already, but I updated the payment summary.

**Comment by ahmedGaber93** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by ahmedGaber93** [comment]
@sakluger BZ checklist is completed!
